tests/element_test_vectors.rs

```diff
@@ -34,60 +37,129 @@ fn contains_path(paths: &[&str], file_name: &str) -> bool {
 // Statically defined arrays containing test file paths to skip in various contexts.
 type SkipList = &'static [&'static str];
 
+// TODO: This trait exists solely for the purposes of this integration test. It allows us to paper
+// over the differences in the ion-c and native Rust ElementWriter implementations.
+// Once all of the tests are passing, we should work to unify their APIs.
+trait RoundTrip {
+    /// Encodes `elements` to a buffer in the specified Ion `format` and then reads them back into
+    /// a `Vec<OwnedElement>` using the provided reader.
+    fn roundtrip<R: ElementReader>(
+        elements: &Vec<OwnedElement>,
+        format: Format,
+        reader: R,
+    ) -> IonResult<Vec<OwnedElement>>;
+}
```

Comment:
🗺️ The `ion-c-sys`-powered writer has some lifetime constraints that make it kind of tough to work with. This trait will enable us to choose which writing implementation we want to run in each test.